nebari_workflow_controller/utils.py

```diff
@@ -88,8 +113,21 @@ def get_keycloak_uid_username(
 
     if label_added_by_argo == "workflows.argoproj.io/creator":
         keycloak_uid = workflow["metadata"]["labels"][label_added_by_argo]
-        keycloak_username = kcadm.get_user(keycloak_uid)["username"]
-        return keycloak_uid, keycloak_username
+        try:
+            keycloak_username = kcadm.get_user(keycloak_uid)["username"]
+            return keycloak_uid, keycloak_username
+        except KeycloakGetError:
+            logger.warning(
+                f"Keycloak user with UID `{keycloak_uid}` not found. Checking if workflow was created by system-serviceaccount..."
+            )
+            preferred_username = workflow["metadata"]["labels"][
+                "workflows.argoproj.io/creator-preferred-username"
+            ]
+            if validate_service_account(keycloak_uid):
+                for user in kcadm.get_users():
+                    if user["username"] == preferred_username:
+                        return user["id"], preferred_username
```

Comment:
This won't work in some cases. You can't always go back to the actual username b/c "Labels only contain [-_.0-9a-zA-Z]", so any other characters will be turned into "-") so not a 1 to 1 mapping.